src/main/java/org/jabref/gui/linkedfile/DownloadLinkedFileAction.java

```diff
@@ -0,0 +1,271 @@
+                                    BibEntry entry,
+                                    LinkedFile linkedFile,
+                                    String downloadUrl,
+                                    DialogService dialogService,
+                                    FilePreferences filePreferences,
+                                    TaskExecutor taskExecutor,
+                                    String suggestedName,
+                                    DoubleProperty downloadProgress) {
+        this.databaseContext = databaseContext;
+        this.entry = entry;
+        this.linkedFile = linkedFile;
+        this.suggestedName = suggestedName;
+        this.downloadUrl = downloadUrl;
+        this.dialogService = dialogService;
+        this.filePreferences = filePreferences;
+        this.taskExecutor = taskExecutor;
+        this.downloadProgress = downloadProgress;
+
+        this.linkedFileHandler = new LinkedFileHandler(linkedFile, entry, databaseContext, filePreferences);
+    }
+
+    public DownloadLinkedFileAction(BibDatabaseContext databaseContext,
+                                    BibEntry entry,
+                                    LinkedFile linkedFile,
+                                    String downloadUrl,
+                                    DialogService dialogService,
+                                    FilePreferences filePreferences,
+                                    TaskExecutor taskExecutor,
+                                    String suggestedName) {
+        this(databaseContext, entry, linkedFile, downloadUrl, dialogService, filePreferences, taskExecutor, suggestedName, null);
+    }
+
+    public DownloadLinkedFileAction(BibDatabaseContext databaseContext,
+                                    BibEntry entry,
+                                    LinkedFile linkedFile,
+                                    String downloadUrl,
+                                    DialogService dialogService,
+                                    FilePreferences filePreferences,
+                                    TaskExecutor taskExecutor,
+                                    DoubleProperty downloadProgress) {
+        this(databaseContext, entry, linkedFile, downloadUrl, dialogService, filePreferences, taskExecutor, "", downloadProgress);
+    }
+
+    public DownloadLinkedFileAction(BibDatabaseContext databaseContext,
+                                    BibEntry entry,
+                                    LinkedFile linkedFile,
+                                    String downloadUrl,
+                                    DialogService dialogService,
+                                    FilePreferences filePreferences,
+                                    TaskExecutor taskExecutor) {
+        this(databaseContext, entry, linkedFile, downloadUrl, dialogService, filePreferences, taskExecutor, "", null);
+    }
+
+    @Override
+    public void execute() {
```

Comment:
As mentioned in the notes of the PR, I could do that (I think nothing much would need be changed in this action for that) but the `LinkedFileViewModel::download` has a slightly different behaviour (which I think might actually be a bug) when the downloaded file is a duplicate: it just forgets the download and leaves the LinkedFile as it was, instead of linking it to the just downloaded file, or the already existing file.

I can definitely change the LinkedFileViewModel::download to use this action, if you think it is okay to be done here.

Reply:
Great, thanks! im a bit busy today, i will tomorrow at it.